channels/jaicp/src/main/kotlin/com/justai/jaicf/channel/jaicp/JaicpConnector.kt

```diff
@@ -31,11 +30,20 @@ abstract class JaicpConnector(
     val channels: List<JaicpChannelFactory>,
     val accessToken: String,
     val url: String,
-    httpClient: HttpClient
+    httpClient: HttpClient,
+    executorThreadPoolSize: Int = DEFAULT_REQUEST_EXECUTOR_THREAD_POOL_SIZE
 ) : WithLogger {
 
+    private val threadPoolRequestExecutor = ThreadPoolRequestExecutor(executorThreadPoolSize)
     private val chatAdapterConnector = ChatAdapterConnector(accessToken, url, httpClient)
     private var registeredChannels = fetchChannels()
+    protected val useLegacyPollingApi: Boolean
+
+    init {
+        useLegacyPollingApi = chatAdapterConnector.getVersion().run {
```

Comment:
Can be replaced by something like: `protected val useLegacyPollingApi = chatAdapterConnector.getVersion() in listOf("release-1.10.1", "release-1.10.2")`